lib/codegen/src/print_errors.rs

```diff
@@ -1,27 +1,68 @@
 //! Utility routines for pretty-printing error messages.
 
 use ir;
+use ir::entities::Inst;
+use ir::function::Function;
 use isa::TargetIsa;
 use result::CodegenError;
+use std::fmt;
 use std::fmt::Write;
 use std::string::{String, ToString};
 use verifier::VerifierError;
+use write::decorate_function;
 
 /// Pretty-print a verifier error.
 pub fn pretty_verifier_error(
     func: &ir::Function,
     isa: Option<&TargetIsa>,
     err: &VerifierError,
 ) -> String {
-    let mut msg = err.to_string();
+    let mut w = String::new();
+    decorate_function(
+        &mut |w, func, isa, inst, indent| pretty_function_error(w, func, isa, inst, indent, err),
+        &mut w,
+        func,
+        isa,
+    ).unwrap();
+    w
+}
+
+/// Pretty-print a function verifier error.
+pub fn pretty_function_error(
```

Comment:
Would this function be useful outside of this file? If not, it'd be good to remove the `pub`.